lib/imgcache.js

```diff
@@ -190,7 +190,10 @@ LOG_LEVEL_ERROR = 3;
   // Returns a URL that can be used to locate a file
   Helpers.EntryGetURL = function (entry) {
     // toURL for html5, toURI for cordova 1.x
-    return (typeof entry.toURL === 'function' ? Helpers.EntryToURL(entry) : entry.toURI());
+    var url = (typeof entry.toURL === 'function' ? Helpers.EntryToURL(entry) : entry.toURI());
+    // Ionic has a URL normalization method for WKWebView (#223)
+    var normalizedUrl = window.Ionic && window.Ionic.normalizeURL && window.Ionic.normalizeURL(url) || url;
```

Comment:
I believe the result could be even more confusing to users because the behavior will be completely different depending on what dependencies are used in the project and part of the behavior of the plugin will be defined elsewhere.

You also just noticed yourself that this integration is apparently bound to break when the Ionic team makes changes to their API. Maintainers of this code should not have to care about Ionic IMHO.

If you're in Angular land, it should be trivial to sanitize your URLs before passing them to imgcache.js, we do so as well in our code. If that is not possible or undesirable, why not pull in the same sanitation code Ionic uses and replicate it in the `Helper` section of the plugin? Then it would work for everyone.

Anyway, I'm not the maintainer of this project. Just wanted to share my thoughts and possibly reach a better solution :)